build.gradle

```diff
@@ -84,6 +86,72 @@ check.doLast {
 }
 /////////////////// END jmh /////////////////////
 
+/////////////////////////////////////////////////////////////////
+//                   SonaType Central Repo                     //
+/////////////////////////////////////////////////////////////////
+/*
+task javadocJar(type: Jar) {
+    classifier = 'javadoc'
+    from javadoc
+}
+*/
+
+task sourcesJar(type: Jar) {
+    classifier = 'sources'
+    from sourceSets.main.allSource
+}
+
+artifacts {
+    archives sourcesJar
+}
+
+signing {
+    sign configurations.archives
```

Comment:
Hi Matt,

Forgive me if I sound confused, but I added a "check" to see if the
qualifications for signing are present, and if not, the uploading will
occur as usual. The problem I thought was with Travis not having the setup
necessary?



On Mon, May 4, 2015 at 1:28 PM, Matthew Taylor <notifications@github.com>
wrote:

> In build.gradle
> <https://github.com/numenta/htm.java/pull/220#discussion_r29609570>:
>
> > +    classifier = 'javadoc'
> > +    from javadoc
> > +}
> > +*/
> > +
> > +task sourcesJar(type: Jar) {
> > +    classifier = 'sources'
> > +    from sourceSets.main.allSource
> > +}
> > +
> > +artifacts {
> > +    archives sourcesJar
> > +}
> > +
> > +signing {
> > +    sign configurations.archives
>
> If you add:
>
>     required { gradle.taskGraph.hasTask("uploadArchives") }
>
> to this closure, people will be able to run gradle build without an
> error. Otherwise, they will get:
>
> Cannot perform signing task ':signArchives' because it has no configured
> signatory
>
> But as long as they are not uploading, they shouldn't be signing either.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numenta/htm.java/pull/220/files#r29609570>.
>



-- 
*With kind regards,*

David Ray
Java Solutions Architect

*cortical.io <http://cortical.io/>*
Sponsor of:  HTM.java <https://github.com/numenta/htm.java>

d.ray@cortical.io
http://cortical.io